### SQL

## The TMDb Database

This Notebook explores the [The Movie Database](https://www.themoviedb.org/): an online movie and TV show database, which houses some of the most popular movies and TV shows at your finger tips. The TMDb database supports 39 official languages used in over 180 countries daily and dates all the way back to 2008. 


<img src="https://github.com/Explore-AI/Pictures/blob/master/sql_tmdb.jpg?raw=true" width=80%/>


Below is an Entity Relationship Diagram (ERD) of the TMDb database:

<img src="https://github.com/Explore-AI/Pictures/blob/master/TMDB_ER_diagram.png?raw=true" width=70%/>

As can be seen from the ER diagram, the TMDb database consists of `12 tables` containing information about movies, cast, genre and so much more.  

For this challenge we will only use the Movies table from the TMDb database. We will apply the use SQL queries to gain insightful information from the Movies table.

Let's get started!

In [ ]:
# Installation needed to work with SQL from the notebook
pip install sqlalchemy
pip install ipython-sql
pip install pymysql

## Loading the database

To begin and start making use of SQL queries you need to prepare your SQL environment you can do this by loading in the magic command `%load_ext sql`, next you can go ahead and load in your database. To do this you will need to ensure you have downloaded the `TMDB.db`sqlite file from Athena and have stored it in a known location. Now that you have all the prerequistes you can go ahead and load it into the notebook. 

In [2]:
%load_ext sql

In [12]:
%%sql
sqlite:///TMDB.db

Throughout this notebook, we will seek to extract information from the database and some key queries we we ask ourselves about the data.

**Question 1**

How many records are there movies table?

In [14]:
%%sql 
SELECT count(*)
FROM movies;

 * sqlite:///TMDB.db
Done.


[(4803,)]

**Solution:**

There are 4803 recordsin the dataset.

**Question 2**

What was the budget for the movie “Inception”?

In [18]:
%%sql
SELECT budget
FROM movies
WHERE title LIKE 'inception%'

 * sqlite:///TMDB.db
Done.


[(160000000,)]

**Solution:**

The budget for the movie “Inception” $160 000 000.

**Question 3**

What is the runtime of the movie "Titanic"?

In [19]:
%%sql
SELECT runtime
FROM movies
WHERE title LIKE 'Titanic%'

 * sqlite:///TMDB.db
Done.


[(194.0,)]

**Solution:**

The runtime of the movie "Titanic" is 194.0.

**Question 4**

How many movies do not have English as their original language?

**Options:**
- 492
- 298
- 387
- 315

In [23]:
%%sql
SELECT count(*)
FROM movies
WHERE original_language NOT LIKE 'en%'

 * sqlite:///TMDB.db
Done.


[(298,)]

**Solution:**

There 298 movies whose original language is not English.

**Question 5**

How many movies are there that have a popularity score of more than 250?

In [24]:
%%sql
SELECT count(*)
FROM movies
WHERE popularity > 250;

 * sqlite:///TMDB.db
Done.


[(7,)]

**Solution:**

There are 7 movies with popularity score of more than 250.

**Question 6**

 How many movies are there where the title is not the same as the original title?
 

In [27]:
%%sql
SELECT count(*)
FROM movies
WHERE title != original_title;

 * sqlite:///TMDB.db
Done.


[(261,)]

**Solution:**

There are 261 movies whose the current title is not the original title.

**Question 7**

 How many movies are there that managed to get a popularity rating of more than 100 with a budget of less than $10 000 000?

In [28]:
%%sql
SELECT count(*)
FROM movies
WHERE popularity > 100 AND budget <  10000000;

 * sqlite:///TMDB.db
Done.


[(5,)]

**Question 8** 

How many movies are there that have the word ‘love’ anywhere in the title?

In [29]:
%%sql
SELECT count(*)
FROM movies
WHERE title LIKE '%love%';

 * sqlite:///TMDB.db
Done.


[(71,)]

**Question 9**

How many movies were released between the dates 1 August 2012 and 31 July 2013?  

In [30]:
%%sql
SELECT count(*)
FROM movies
WHERE release_date > '2012-08-01' AND release_date < '2013-07-31';

 * sqlite:///TMDB.db
Done.


[(227,)]

**Solution:**

227 movies were released between the dates 1 August 2012 and 31 July 2013.

**Question 10**

You have had a long day and want to sit back and enjoy a movie.  Unfortunately, today you are only in the mood for a very specific type of movie. It definitely needs to be in English. It should also be new, something after 1 Jan 2010, but not too new as you might have seen it recently, so it must have been released before 1 Jan 2016. It should also be a romantic movie, so make sure it has the word love somewhere in the title.  You also want it to be a big blockbuster, so the budget of the move must be more than $10 000 000.

What is the movie with the highest popularity that meets all of your requirements?

**Options:**
-  Love & Other Drugs
- From Paris with Love
-  Crazy, Stupid, Love
- Eat Pray Love

In [47]:
%%sql
SELECT max(popularity) as 'most_popular', title
FROM movies
WHERE original_language LIKE 'en' AND release_date > '2010-01-01' AND release_date < '2016-01-1' AND title LIKE '%love%' AND budget >  10000000

 * sqlite:///TMDB.db
Done.


[(37.990705, 'Crazy, Stupid, Love.')]